### PR TITLE
add twitter verification proof

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -19,13 +20,12 @@ const mainContainer = classnames(
...
+      <Founder />
       <Root>
         <Router hook={useHashLocation}>
           <Switch>
-            <Route component={MainBlock} path="/allow-map" />
-            <Route path="">
-              <Redirect to="/allow-map" />
-            </Route>
+            <Route path="/allow-map" component={MainBlock} />
+            <Route path="/yc" component={Founder} />
```
Comment (T-Damer) [inline src/App.tsx]: Do we still need redirect from any other route to `/allow-map`?

File: src/components/Founder.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+)
+const errorText = classnames(
+  textColor('text-red-500'),
+  textAlign('text-center')
+)
+
+export default function () {
+  useEffect(() => {
+    const handleMessage = async (message: unknown) => {
+      if (typeof message !== 'object' || !message || !('data' in message))
```
Comment (T-Damer) [inline src/components/Founder.tsx]: Extract this into helper `isDataInMessage`

File: src/components/Founder.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        return
+      const { data } = message as { data: string }
```
Comment (T-Damer) [inline src/components/Founder.tsx]: You'll not need `as { data: string }` if you'll create a helper that checks the type

File: src/components/Founder.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      try {
+        const proof = await createFounderProof(data)
+        postWebViewMessage({
+          data,
+          type: Messages.GetTwitterProof,
+        })
+      } catch {}
```
Comment (T-Damer) [inline src/components/Founder.tsx]: Let's log something at least

File: src/components/Founder.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    if (navigator.userAgent.includes('Android')) {
+      document.addEventListener('message', handleMessage)
```
Comment (T-Damer) [inline src/components/Founder.tsx:46]: I assume that you can use `document` for both platforms

File: src/helpers/createFounderProof.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+export async function getEddsaPublicKey() {
+  const { data } = await axios.get<PublicKey>(
+    `${baseURL}/verify/eddsa-public-key`
+  )
+  return data
+}
+
+export default async function (token: string) {
+  try {
+    const response = await fetch('/trees/yc/1.json')
```
Comment (T-Damer) [inline src/helpers/createFounderProof.ts]: It fetches it's own file? Can we simply do `require` here?

Comment (T-Damer) [reply src/helpers/createFounderProof.ts]: Or maybe we can import this file?

File: src/App.tsx [hunk 7]
```diff
@@ -19,13 +20,12 @@ const mainContainer = classnames(
 export default function () {
```
Comment (T-Damer) [inline src/App.tsx]: Now let's name default exports in every project, so we can debug better in future

File: src/helpers/getYCMerkleTreeProof.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import { BigNumber } from 'ethers'
+import { IncrementalMerkleTree } from '@zk-kit/incremental-merkle-tree'
+import { buildPoseidon } from 'circomlibjs'
+
+export async function getYCMerkleTreeInputs(
+  hashFunc: (values: string[]) => string | number | bigint | boolean,
+  commitment: bigint | string,
```
Comment (T-Damer) [inline src/helpers/getYCMerkleTreeProof.ts]: Let's create a type for `Commitment`

File: src/helpers/createFounderProof.ts (hunk 6 above)
Comment (MixailE) [reply src/helpers/createFounderProof.ts]: In theory, you can only fetch from the public folder

File: src/App.tsx (hunk 1 above)
Comment (MixailE) [reply src/App.tsx]: Yes

File: src/components/Founder.tsx (hunk 5 above)
Comment (MixailE) [reply src/components/Founder.tsx:46]: We will test this a bit later

File: src/components/Founder.tsx [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  space('space-y-2'),
+  padding('px-8'),
+  gap('gap-2')
+)
+
+export default function Founder() {
+  useEffect(() => {
+    const handleMessage = (message: unknown) => {
+      if (!isDataInMessage(message)) return
+      const { data } = message as {
```
Comment (T-Damer) [inline src/components/Founder.tsx:33]: We should make sure there's something in `data`

Comment (T-Damer) [reply src/components/Founder.tsx:33]: We should make sure that `data` contains everything we need, without checking it we might end up having corrupted data here

File: src/components/Founder.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+      )
+    }
+
+    if (navigator.userAgent.includes('Android')) {
+      document.addEventListener('message', handleMessage)
+    } else {
+      window.addEventListener('message', handleMessage)
+    }
+    return () => {
+      window.onmessage = handleMessage
```
Comment (T-Damer) [inline src/components/Founder.tsx:51]: This should happen if environment isn't Android?

Comment (T-Damer) [reply src/components/Founder.tsx:51]: Because Android doesn't have window?

Comment (MixailE) [reply src/components/Founder.tsx:51]: It's made on purpose

File: src/components/Founder.tsx (hunk 9 above)
Comment (MixailE) [reply src/components/Founder.tsx:33]: We can add validation and error output later

**Review by T-Damer — APPROVED** [review]

**Pull request merged**